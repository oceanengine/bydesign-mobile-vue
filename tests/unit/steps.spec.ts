import { mount, later } from '../index';
import { Steps as StepsNative, Step as StepNative } from '@src';
const Step = StepNative as any;
const Steps = StepsNative as any;

describe('Steps', () => {
  test('horizontal step', async () => {
    const wrapper = mount({
      template: `
        <byted-steps :current="current">
        <byted-step v-for="item in steps" :key="item.title" v-bind="item"></byted-step>
        </byted-steps>
      `,
      components: {
        'byted-steps': Steps,
        'byted-step': Step,
      },
      methods: {},
      data() {
        return {
          current: 1,
          steps: [
            {
              title: '已完成',
            },
            {
              title: '进行中',
              inProgress: true,
            },
            {
              title: '待完成',
            },
            {
              title: '未完成',
              status: 'error',
            },
          ],
        };
      },
    });

    await later();
    expect(wrapper).toMatchSnapshot();
  });

  test('event change', async () => {
    const onChange = jest.fn();
    const wrapper = mount({
      template: `
        <byted-steps :current="1" @change="onChange">
          <byted-step v-for="item in steps" :key="item.title" v-bind="item"></byted-step>
        </byted-steps>
      `,
      components: {
        'byted-steps': Steps,
        'byted-step': Step,
      },
      methods: {
        onChange,
      },
      data() {
        return {
          current: 1,
          steps: [
            {
              title: '已完成',
            },
            {
              title: '进行中',
              inProgress: true,
            },
            {
              title: '待完成',
            },
            {
              title: '未完成',
              status: 'error',
            },
          ],
        };
      },
    });

    await later();

    const items = wrapper.findAll('.byted-step');
    items.at(1).trigger('click');

    expect(onChange).toBeCalledTimes(1);
  });
});
