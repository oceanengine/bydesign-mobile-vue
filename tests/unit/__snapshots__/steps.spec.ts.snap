// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps horizontal step 1`] = `
<div class="byted-steps byted-steps--horizontal byted-steps--label-horizontal">
  <div class="byted-step byted-step--horizontal byted-step--md byted-step--finished byted-step--active">
    <div class="byted-step__head"><i class="byted-icon byted-icon--check-one byted-step__icon">test-file-stub</i><span class="byted-step__title">已完成</span></div>
    <div class="byted-step__split-panel"></div>
  </div>
  <div class="byted-step byted-step--horizontal byted-step--md byted-step--inProgress">
    <div class="byted-step__head"><span class="byted-step__icon-number">2</span><span class="byted-step__title">进行中</span></div>
    <div class="byted-step__split-panel"></div>
  </div>
  <div class="byted-step byted-step--horizontal byted-step--md byted-step--waiting">
    <div class="byted-step__head"><span class="byted-step__icon-number">3</span><span class="byted-step__title">待完成</span></div>
    <div class="byted-step__split-panel"></div>
  </div>
  <div class="byted-step byted-step--horizontal byted-step--md byted-step--error">
    <div class="byted-step__head"><span class="byted-step__icon-number">4</span><span class="byted-step__title">未完成</span></div>
  </div>
</div>
`;
